pkg/apis/apps/v1beta2/defaults_test.go

```diff
@@ -416,13 +406,6 @@ func TestSetDefaultReplicaSet(t *testing.T) {
 			t.Errorf("unexpected object: %v", rs2)
 			t.FailNow()
 		}
-		if test.expectSelector != reflect.DeepEqual(rs2.Spec.Selector.MatchLabels, rs2.Spec.Template.Labels) {
```

Comment:
> Do not default the selector to anything. 
Did you mean `obj.Spec.Selector` should be `nil`? `obj.Spec.Selector` will be `nil` without explicitly setting it.